src/app/checklist/[checklist_slug]/[locale]/page.tsx

```diff
@@ -0,0 +1,138 @@
+import { Avatar } from "@/components/avatar";
+import {
+    Checklist as IChecklist,
+    ChecklistService,
+} from "@/services/checklist";
+import { Tags } from "@/components/tag";
+import { CircleStackIcon } from "@heroicons/react/24/solid";
+import { Checklist } from "@/components/checklist";
+import { LocalesDropdown } from "@/components/localeDropdown";
+import Link from "next/link";
+import { Metadata } from "next";
+import Script from "next/script";
+
+interface ChecklistParam {
+    checklist_slug: string;
+    locale: string;
+}
+
+// Generate all static paths that lead to checklists on the format /<checklist_slug>/<locale>
+export const generateStaticParams = (): ChecklistParam[] => {
+    const checklistPaths = ChecklistService.getInstance().getChecklistFiles();
+
+    return checklistPaths.map(({ slug, locale }) => ({
+        checklist_slug: slug,
+        locale,
+    }));
+};
+
+// Generate the page metadata
+export async function generateMetadata({
+    params: { checklist_slug: slug, locale },
+}: {
+    params: ChecklistParam;
+}): Promise<Metadata> {
+    const checklist = ChecklistService.getInstance().getChecklistData(
+        slug,
+        locale
+    );
+
+    return {
+        title: `DevChecklists | ${checklist.frontmatter.title ?? ""}`,
+        description: checklist.frontmatter.description,
+        keywords: checklist.frontmatter.tags,
+    };
+}
+
+// Used to provide rich results for search engines
+// https://developers.google.com/search/docs/appearance/structured-data/how-to
+function generateJSONLD(checklist: IChecklist) {
+    return {
+        __html: JSON.stringify({
+            "@context": "https://schema.org",
+            "@type": "HowTo",
+            name: checklist.frontmatter.title ?? checklist.slug,
+            // TODO: Extract checkitems and add them to JSON-LD
+            step: [],
+        }),
+    };
+}
+
+const ChecklistPage = ({
+    params: { checklist_slug: slug, locale },
+}: {
+    params: ChecklistParam;
+}) => {
+    const checklist = ChecklistService.getInstance().getChecklistData(
+        slug,
+        locale
+    );
+    const {
+        frontmatter: {
+            title,
+            description,
+            author_username,
+            author_name,
+            tags = [],
+        },
+        availableLocales,
+    } = checklist;
+
+    return (
+        <>
+            <Script
+                id="ld-json"
+                type="application/ld+json"
+                dangerouslySetInnerHTML={generateJSONLD(checklist)}
```

Comment:
Is this the recommended way to set JSONLD with NextJS?

Reply:
I know this isn't unsafe, due to the JSON stringify and nature of data.

Reply:
Yes
https://nextjs.org/learn/seo/rendering-and-ranking/metadata